src/ElmFormat/Render/Box.hs

```diff
@@ -2121,6 +2065,87 @@ formatComment comment =
             mustBreak $ row [ punc "{--", literal c, punc "-}" ]
 
 
+formatCaseExpression ::
+    ElmVersion
+    -> ImportInfo
+    -> (AST.Commented AST.Expression.Expr, AST.Multiline)
+    -> ([(AST.Commented AST.Pattern.Pattern, (AST.Comments, AST.Expression.Expr))], AST.Multiline)
+    -> Box
+formatCaseExpression elmVersion importInfo (subject, multilineSubject) (branches, multilineFirstBranch) =
+    let
+        opening =
+          case
+            ( multilineSubject
+            , formatCommentedExpression elmVersion importInfo SyntaxSeparated subject
+            )
+          of
+              (AST.JoinAll, SingleLine subject') ->
+                  line $ row [ keyword "case" , space , subject' , space , keyword "of" ]
+              (_, subject') ->
+                  stack1
+                      [ line $ keyword "case"
+                      , indent subject'
+                      , line $ keyword "of"
+                      ]
+
+        (multilineBranches, branches') =
+            List.mapAccumR
+            (\mline (AST.Commented prePat pat postPat, (preRes, res)) ->
+                let
+                    prePat' =
+                        Maybe.maybeToList $ formatComments prePat
+                    pat' =
+                        formatPattern elmVersion False pat
+                            |> negativeCasePatternWorkaround pat
+                    postPat' =
+                        Maybe.maybeToList $ formatComments postPat
+                    preRes' =
+                        Maybe.maybeToList $ formatComments preRes
+                    res' =
+                        formatExpression elmVersion importInfo SyntaxSeparated res
```

Comment:
`res`/`res'`/`preRes` should I think be called "body" instead of "res" to be consistent with other code, unless there was a specific reason you chose that name.

Reply:
Fixed